esp-hal/src/spi/master.rs

```diff
@@ -3444,10 +3449,19 @@ impl Driver {
 
     #[cfg_attr(place_spi_driver_in_ram, ram)]
     async fn transfer_in_place_async(&self, words: &mut [u8]) -> Result<(), Error> {
+        // Cut the transfer short if the future is dropped. We'll block for a short
+        // while to ensure the peripheral is idle.
+        let cancel_on_drop = OnDrop::new(|| {
+            // Commented for now to make sure the added test fails.
+            // self.abort_transfer();
+            // while self.busy() {}
+        });
+
         for chunk in words.chunks_mut(FIFO_SIZE) {
             self.write_async(chunk).await?;
             self.read_from_fifo(chunk)?;
         }
+        cancel_on_drop.defuse();
```

Comment:
Can I ask you to move the `cancel_on_drop` into the for loop to wrap the `self.write_async(chunk).await` specifically?

Something like this:
```Rust
let cancel_on_drop = OnDrop::new(|| ....);
let result = self.write_async(chunk).await;
cancel_on_drop.defuse();
result?;
```

It gives me less to worry about when reviewing the code 😄